### ISSUE TITLE
vscode.Uri.file is not a constructor

### ISSUE THREAD

**Issue by furier** [Gruntfuggly/todo-tree#125]
When I try to open the extension this error message pops up in VS Code. `vscode.Uri.file is not a constructor`.

![image](https://user-images.githubusercontent.com/776919/52624653-7fcff380-2eaf-11e9-8d4f-5960e1134ca9.png)

I am using the insider version: `1.32.0-insider`

**Comment by Gruntfuggly** [comment]
Please could you have a look in the developer console and see if there is a call stack with a line number?

**Comment by joseph-turner** [comment]
I'm also getting this error. Here's the full error stack:
```
[error] vscode.Uri.file is not a constructor: TypeError: vscode.Uri.file is not a constructor
    at TreeNodeProvider.getTreeItem (/Users/jturner/.vscode-insiders/extensions/gruntfuggly.todo-tree-0.0.118/tree.js:347:40)
    at define.resolveTreeNode.a.asPromise (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:430:89)
    at t.asPromise.Promise (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:43:678)
    at new Promise (<anonymous>)
    at Object.t.asPromise (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:43:652)
    at g.resolveTreeNode (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:430:57)
    at define.reveal.function.dataProvider.getParent.refreshPromise.then.then.t (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:428:899)
```

**Comment by Gruntfuggly** [comment]
Hmm - I'm not entirely sure whats going on? `vscode.Uri.file` is a standard API call:
https://code.visualstudio.com/api/references/vscode-api#Uri
Maybe they are trying some changes to the tree view API in the insiders build. Can you let me know if it still occurs with later versions of insiders?

**Comment by 123swk123** [comment]
Here is my version info,

> Version: 1.32.0-insider
> Commit: a509265116babc17648c9052eeeb8ae2909fed87
> Date: 2019-02-13T06:14:01.838Z
> Electron: 3.1.3
> Chrome: 66.0.3359.181
> Node.js: 10.2.0
> V8: 6.6.346.32
> OS: Linux x64 4.20.7-042007-generic

creates same error, But the same insider version on windows works well.

**Comment by furier** [comment]
@123swk123 what do you mean? Im on windows and it doesn't work.

**Comment by 123swk123** [comment]
ii use insider version 1.32.0 on both linux and windows. i see this behavior only with linux not on windows.

**Comment by Gruntfuggly** [comment]
I don't know why it fails using insiders, but you can work around it be disabling `todo-tree.showBadges`.

**Comment by furier** [comment]
@Gruntfuggly that worked thnxs :)

**Comment by stonesbg** [comment]
This fixed the issue for me as well.  Also using vscode-insiders. Thanks

**Comment by Gruntfuggly** [comment]
Fixed in the latest version - see #133
